robot/Cumulus/tests/browser/relationships/create_relationships.robot

```diff
@@ -1,48 +1,60 @@
+    Run Keyword If Test Failed  Capture Page Screenshot
+
+Setup Test Data
+    &{contact1} =                        API Create Contact    Email=automation@example.com
+    Store Session Record                 Account               &{contact1}[AccountId]
+    Set suite variable                   &{contact1}
+    &{contact2} =                        API Create Contact
+    Store Session Record                 Account               &{contact2}[AccountId]
+    Set suite variable                   &{contact2}
 
 *** Test Cases ***
 
 Create Relationships for contacts
-    [tags]  unstable
-    &{contact1} =  API Create Contact    Email=skristem@robot.com
-    Store Session Record    Account    &{contact1}[AccountId]
-    Go To Record Home  &{contact1}[AccountId]
-    
-    #2 Create a new contact under HouseHold Validation
-    ${contact_id2} =  New Contact for HouseHold
-    &{contact2} =  Salesforce Get  Contact  ${contact_id2}
-    Store Session Record    Account    &{contact2}[AccountId]
-    Page Should Contain  &{contact1}[LastName] and &{contact2}[LastName] Household
-    Select Tab  Related
-    Click Related List Button  Relationships    New
-    Populate Lookup Field    Related Contact    &{contact1}[FirstName] &{contact1}[LastName]
-    Select Value From Dropdown   Type              Parent
-    Click Modal Button        Save
-    ${expected result1}        Catenate    &{contact1}[FirstName] &{contact1}[LastName]    is    &{contact2}[FirstName] &{contact2}[LastName]'s    Parent
-    Log to Console    ${expected result1}
-    ${id}    ${status1}    Check Status    &{contact1}[FirstName] &{contact1}[LastName]
-    Should Be Equal As Strings    ${status1}    ${expected result1}
-    Click Link    link=Show more actions
-    Click Link    link=Relationships Viewer
-    Sleep    5 
+    [Documentation]                      Create 2 contacts with API which inturn creates 2 household accounts.
+    ...                                  Open contact2 record and create a new relationship with contact 1.
+    ...                                  Verify the parent-child relationship is correctly established.
+
+    [tags]                               W-037650                                             feature:Relationships
+
+    Go To Page                           Details                                              Contact                                               object_id=&{contact2}[Id]
+    Select Tab                           Related
+    Wait Until Loading Is Complete
+    Click Related List Button            Relationships                                        New
+    Wait For Modal                       New                                                  Relationship
+
+    Populate Lookup Field                Related Contact                                      &{contact1}[FirstName] &{contact1}[LastName]
+    Select Value From Dropdown           Type                                                 Parent
```

Comment:
can you try to use populate modal form keyword whenever u open a modal? If that doesn't work then try the keywords directly